### ISSUE TITLE
Document usage of optionsJson in PutMarkLogic

### ISSUE THREAD

**Issue by rjrudin** [marklogic/nifi#194]
`optionsJson` is at this point a deprecated technique for capturing the URIs in a PutML batch so that they can be used with DHF. However, the structure of the JSON options is intended for use with an internal DHF endpoint, which we don't want to encourage use of. Additionally, there are no docs for this - although it's visible to a user when using e.g. `LogAttribute` to see what's in the FlowFile.

I think we need two enhancements here:

1. Use `WritesAttribute` on PutML to document the existence of this attribute and what its purpose is
2. Consider an enhancement that constructs the options JSON that a user would actually want with DHF, which is one that uses `sourceQuery` with e.g. a cts.documentQuery for constraining on the written URIs. We likely would want a user to turn this feature on, since it has no value unless the user wants to run a flow.

**Comment by rjrudin** [comment]
Until then, a user can adapt the `optionsJson` value into a proper set of DHF options that overrides the `sourceQuery` by inserting an `ExecuteScript` processor in between PutML and RunFlowML with the following ecmascript body:

```
flowFile = session.get();
uris = JSON.parse(flowFile.getAttribute("optionsJson")).uris;
sourceQuery = "cts.documentQuery(['" + uris.join("','") + "'])";
dhfOptions = '{"sourceQuery": "' + sourceQuery + '"}';
session.putAttribute(flowFile, "dhfOptions", dhfOptions);
session.transfer(flowFile, REL_SUCCESS);
```

**Comment by rjrudin** [comment]
We're planning the 1.16.3.3 release soon, and given that this 1) does not work as expected and requires the above script to be useful, and 2) the above script can easily be used with the existing `URIs` attribute which stores the URIs as a comma-delimited list, we are going to consider `optionsJson` to be broken and unnecessary and are going to remove it. 

Below is a script equivalent to what's above, but simply using the `URIs` attribute instead:

```
flowFile = session.get();
uris = flowFile.getAttribute("URIs");
sourceQuery = "cts.documentQuery(['" + uris.split(",").join("','") + "'])";
dhfOptions = '{"sourceQuery": "' + sourceQuery + '"}';
session.putAttribute(flowFile, "dhfOptions", dhfOptions);
session.transfer(flowFile, REL_SUCCESS);
```

**Comment by rjrudin** [comment]
Resolved via #199
